src/main/java/reposense/commits/CommitResultAggregator.java

```diff
@@ -39,60 +44,74 @@ public static CommitContributionSummary aggregateCommitResults(
+    private static float getContributionVariance(List<AuthorDailyContribution> contributions,
+            Date startDate, Date lastDate) {
         if (contributions.size() == 0) {
             return 0;
         }
         //get mean
         float total = 0;
+        long totalDays = (lastDate.getTime() - getStartOfDate(startDate).getTime())
+                / DAYS_IN_MS + 1;
+
         for (AuthorDailyContribution contribution : contributions) {
             total += contribution.getTotalContribution();
         }
-        float mean = total / contributions.size();
+        float mean = total / totalDays;
+
         float variance = 0;
-        for (AuthorDailyContribution contribution : contributions) {
-            variance += Math.pow((mean - contribution.getTotalContribution()), 2);
+        long currentDate = getStartOfDate(startDate).getTime();
+        int contributionIndex = 0;
+        for (int i = 0; i < totalDays; i += 1) {
+            if (contributionIndex < contributions.size()
+                    && currentDate == contributions.get(contributionIndex).getDate().getTime()) {
+                variance += Math.pow((mean - contributions.get(contributionIndex).getTotalContribution()), 2);
+                contributionIndex += 1;
+            } else {
+                variance += Math.pow(mean, 2);
+            }
+            currentDate += DAYS_IN_MS;
         }
-        return variance / contributions.size();
+        return variance / totalDays;
     }
 
     private static Map<Author, List<AuthorDailyContribution>> getAuthorDailyContributionsMap(
-            Set<Author> authorSet, List<CommitResult> commitResults, Date startDate) {
+            Set<Author> authorSet, List<CommitResult> commitResults) {
         Map<Author, List<AuthorDailyContribution>> authorDailyContributionsMap = new HashMap<>();
         authorSet.forEach(author -> authorDailyContributionsMap.put(author, new ArrayList<>()));
 
-        Date currentDate = getStartOfDate(startDate);
-        Date nextDate = getNextDayDate(currentDate);
-        addDailyContributionForNewDate(authorDailyContributionsMap, currentDate);
-
+        Date commitStartDate = null;
         for (CommitResult commitResult : commitResults) {
-            while (nextDate.before(commitResult.getTime())) {
-                currentDate = new Date(nextDate.getTime());
-                nextDate = getNextDayDate(nextDate);
-                addDailyContributionForNewDate(authorDailyContributionsMap, currentDate);
+            commitStartDate = getStartOfDate(commitResult.getTime());
+            Author commitAuthor = commitResult.getAuthor();
+
+            List<AuthorDailyContribution> authorDailyContributions =
+                    authorDailyContributionsMap.get(commitAuthor);
+
+            if (authorDailyContributions.isEmpty()
+                    || !authorDailyContributions.get(authorDailyContributions.size() - 1)
+                            .getDate().equals(commitStartDate)) {
```

Comment:
Let's put the `.getDate()` with the line on top, so that its easier to see that the entire top line is the 1st date value, and the next line is the 2nd date to be compared to.